controlplane/kubeadm/internal/controllers/controller_test.go

```diff
@@ -824,10 +1091,16 @@ func TestKubeadmControlPlaneReconciler_ensureOwnerReferences(t *testing.T) {
 
 	t.Run("replace non-KCP controller with KCP controller reference", func(t *testing.T) {
 		objs := []client.Object{fakeGenericMachineTemplateCRD, cluster.DeepCopy(), kcp.DeepCopy(), tmpl.DeepCopy()}
-		for _, purpose := range []secret.Purpose{secret.ClusterCA, secret.FrontProxyCA, secret.ServiceAccount, secret.EtcdCA} {
+		certificates := secret.Certificates{
```

Comment:
Let's definitely preserve those changes for envtest on a branch/PR somewhere, so we can use them later for the envtest migration